pkg/spireentry/reconciler.go

```diff
@@ -630,3 +640,31 @@ func idsFromEntries(entries []spireapi.Entry) []string {
 	}
 	return ids
 }
+
+// filterJoinTokenEntries filters out entries that correspond to join tokens.
+func filterJoinTokenEntries(entries []spireapi.Entry) []spireapi.Entry {
+	var filteredEntries []spireapi.Entry
```

Comment:
nit: we should take the opportunity to pre-allocate the slice to reduce allocation pressure
```suggestion
	if len(entries) == 0 {
		return entries
	}	
	filteredEntries := make([]spireapi.Entry, 0, len(entries))
```